### PR TITLE
Remove unneeded permission in auto-merge example

### PR DESCRIPTION
**Pull request by felipecrs**
And simplify the workflow a little.

### Why:

Closes [issue link]

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by felipecrs** [comment]
I tested this in https://github.com/vscode-shellcheck/vscode-shellcheck/blob/master/.github/workflows/dependabot-auto-merge.yaml

**Comment by ramyaparimi** [comment]
@felipecrs 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by felicitymay** [comment]
@ramyaparimi - since this deals with actions and permissions, this will need an SME review from either actions and/or Dependabot.

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 1]
```diff
@@ -465,21 +465,20 @@ name: Dependabot auto-merge
...
 
 permissions:
-  pull-requests: write
   contents: write
 
 jobs:
   dependabot:
     runs-on: ubuntu-latest
-    if: ${{ github.actor == 'dependabot[bot]' }}
+    if: github.actor == 'dependabot[bot]'
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:473]:
```suggestion
    if: ${{ github.actor == 'dependabot[bot]' }}
```

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 2]
```diff
@@ -465,21 +465,20 @@ name: Dependabot auto-merge
...
     steps:
       - name: Dependabot metadata
         id: dependabot-metadata
         uses: dependabot/fetch-metadata@v1.1.1
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Enable auto-merge for Dependabot PRs
-        if: ${{contains(steps.dependabot-metadata.outputs.dependency-names, 'my-dependency') && steps.dependabot-metadata.outputs.update-type == 'version-update:semver-patch'}}
+        if: contains(steps.dependabot-metadata.outputs.dependency-names, 'my-dependency') && steps.dependabot-metadata.outputs.update-type == 'version-update:semver-patch'
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:481]:
```suggestion
        if: ${{contains(steps.dependabot-metadata.outputs.dependency-names, 'my-dependency') && steps.dependabot-metadata.outputs.update-type == 'version-update:semver-patch'}}
```

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 3]
```diff
@@ -497,21 +496,20 @@ name: Dependabot auto-merge
...
 
 permissions:
-  pull-requests: write
   contents: write
 
 jobs:
   dependabot:
     runs-on: ubuntu-latest
-    if: ${{ github.actor == 'dependabot[bot]' }}
+    if: github.actor == 'dependabot[bot]'
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:504]:
```suggestion
    if: ${{ github.actor == 'dependabot[bot]' }}
```

File: content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md [hunk 4]
```diff
@@ -497,21 +496,20 @@ name: Dependabot auto-merge
...
     steps:
       - name: Dependabot metadata
         id: metadata
         uses: dependabot/fetch-metadata@v1.1.1
         with:
           github-token: "${{ secrets.GITHUB_TOKEN }}"
       - name: Enable auto-merge for Dependabot PRs
-        if: ${{contains(steps.metadata.outputs.dependency-names, 'my-dependency') && steps.metadata.outputs.update-type == 'version-update:semver-patch'}}
+        if: contains(steps.metadata.outputs.dependency-names, 'my-dependency') && steps.metadata.outputs.update-type == 'version-update:semver-patch'
```
Comment (mchammer01) [inline content/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions.md:512]:
```suggestion
        if: ${{contains(steps.metadata.outputs.dependency-names, 'my-dependency') && steps.metadata.outputs.update-type == 'version-update:semver-patch'}}

**Review by mchammer01 — COMMENTED** [review]
@felipecrs - thanks for your contribution ✨ 
I've accepted the change about the permission but will revert the updates made to `${{` following consultation with the Dependabot team. You are right in that we could remove these but most of our examples throughout the docs contain them so we'll keep them for consistency for now. 
Thanks again for helping us improve our docs!

**Review by mchammer01 — APPROVED** [review]
LGTM :shipit:

**Comment by felipecrs** [comment]
@mchammer01 it makes sense. Thank you!

**Pull request merged**